ext/dom/document.c

```diff
@@ -1484,42 +1485,57 @@ PHP_METHOD(DOMDocument, saveXML)
 			php_dom_throw_error(WRONG_DOCUMENT_ERR, dom_get_strict_error(intern->document));
 			RETURN_FALSE;
 		}
+
 		buf = xmlBufferCreate();
 		if (!buf) {
 			php_error_docref(NULL, E_WARNING, "Could not fetch buffer");
 			RETURN_FALSE;
 		}
-		if (options & LIBXML_SAVE_NOEMPTYTAG) {
-			saveempty = xmlSaveNoEmptyTags;
-			xmlSaveNoEmptyTags = 1;
-		}
+		saveempty = xmlSaveNoEmptyTags;
+		xmlSaveNoEmptyTags = !!(options & LIBXML_SAVE_NOEMPTYTAG);
```

Comment:
```suggestion
		xmlSaveNoEmptyTags = (options & LIBXML_SAVE_NOEMPTYTAG);
```
I'm confused by the double ``!!`` here

Reply:
We must put 1 in `xmlSaveNoEmptyTags` is `LIBXML_SAVE_NOEMPTYTAG` is set in options. `(options & LIBXML_SAVE_NOEMPTYTAG)` does not equal 1 because `LIBXML_SAVE_NOEMPTYTAG` is not 1. By using `!!` the value is converted to a bool basically.
I can change it to a ternary `(options & LIBXML_SAVE_NOEMPTYTAG) ? 1 : 0` though.